Report/sections/4_recapitulation/discussion.tex

```diff
@@ -0,0 +1,45 @@
+\chapter{Discussion}
+Our goal with this project was to design and implement a compiler for a programming language for beginners in the context of game development. 
+This sections seeks to communicate the circumstances under which the project was developed as well as give insight into our thoughts about the end result.
+We also wish to elaborate on specific decisions that we, in retrospect, feel were poor or overly ambitious.
+
+\section{Ambitiousness}
+A major challenge in the first half of the project that ended up impacting the project as a whole was that our original group had a difference in ambition. 
+One half of the group had lower expectations which caused conflicts and limited the group work.
+Much time was spent attemping to solve these conflicts and getting everyone to agree on how much effort and time to invest in the project.
+This combined with the overall level of ambition meant that we fell behind as not much work was being done on the project for the first two months. 
+Eventually, this resulted in the decision to split our group in two halves as we were unable to find an agreeable compromise.
+
+Because we stuck to the original vision for the project despite having less time and a smaller group, we unfortunately ended up investing too much time into features that we realistically did not have time to implement such as member access.
+Features like this and others were not part of the semester curriculum which therefore meant that we had no frame of reference to work from.
+
+Furthermore, the consequence of the group split meant that we were unable to spend as much time as we would have liked on the game aspect of \dazel{}. 
+Thus we ended with a relatively simple prototype, however we feel that, under the circumstances, this is a satisfying proof of concept.
+
+\section{Learning Goals}
+Part of the semester learning goals state that we should be able to demonstrate a working knowledge of all the phases of a compiler or interpreter. We feel that we succeded in achieving this, but in hindsight it is very clear that there are areas in which we, as previously mentioned, may have been too ambitious. 
+
+One particular feature that stands out as an immense technical challenge has been the implementation of member access. A large part of the difficulty with this feature has been finding helpful literature on how one could go about implementing such a feature and as a result a lot of time was spent making a solution that worked as one would expect. The learning experience has been great, but because of the fact that such a feature does not contribute towards any of the semester learning goals, we feel that time may have been better spent on improving or implementing things that contribute more greatly towards those learning goals. 
```

Comment:
The last few sentences may be... dangerous?

I get the reflective nature of them, but nevertheless, they might do more harm than we gain.

Member access also teaches us about implementing semi-OOPL features, which might contribute to a better understanding of creating compilers for languages under that paradigm.

SPO teaches us about language design. It teaches us that design is _very_ important. Perhaps this feature was, in some sense, very valuable to us. Not just in some measure of learning experiences, but also in designing a language that we feel is useful to users. Sure, code generation could have been expanded to include more things. But the core design of something than a few days of coding work. It's much harder to change the core of something later than to implement new, cool things.